x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/SqlProtocolTestCase.java

```diff
@@ -116,6 +116,60 @@ public void testDateTimeIntervals() throws IOException {
         assertQuery("SELECT INTERVAL '163:59.163' MINUTE TO SECOND", "INTERVAL '163:59.163' MINUTE TO SECOND", "interval_minute_to_second",
                 "PT2H43M59.163S", "+0 02:43:59.163", 23);
     }
+    
+    /**
+     * Method that tests that a binary response (CBOR) will return either Float or Double, depending on the SQL data type, for floating
```

Comment:
Would it be worth adding half-float type tests?
If the mapping contains them, the driver will probably get it as such.
The conversion doesn't work with this type, though, so not sure if that'd be easily doable.

Reply:
I don't think it's possible to create a sql query with literals that would return a half_float. The only possibility would be to create an index that has a `half_float` data type field in it. That would be part of another class but I don't think it's worth adding it as part of this PR.